packages/vaadin-overlay/test/focus-trap.test.js

```diff
@@ -105,18 +106,6 @@ describe('focus-trap', function () {
       expect(focusableElements[9]).to.eql(overlay.content.querySelector('vaadin-button'));
     });
 
-    it('should not focus next focusable element inside the content when Tab is prevented', async () => {
-      overlay.addEventListener('keydown', (event) => {
-        if (event.key === 'Tab') {
-          event.preventDefault();
-        }
-      });
-      overlay.opened = true;
-      await oneEvent(overlay, 'vaadin-overlay-open');
-      tabKeyDown(focusableElements[overlay._focusedIndex()]);
-      expect(overlay._focusedIndex()).to.equal(0);
-    });
-
```

Comment:
It was previously possible to prevent the focus trap logic on `keydown` by calling `preventDefault()` on the event (see [the check](https://github.com/vaadin/web-components/pull/3159/files#diff-a22b589817f0a0a072a03603aacbf01899a28200af52529f4d22879f213a5776L503)). The workaround was introduced as a fix for vaadin/vaadin-rich-text-editor#125.

However, it appears that over time the mentioned issue was fixed directly in the rich text editor library as I wasn't able to reproduce it even without the `defaultPrevented` workaround. Considering this, I decided to drop the workaround at all.